src/merkle_tree/mod.rs

```diff
@@ -234,26 +239,145 @@ impl<P: Config> MerkleTree<P> {
     }
 
     /// Returns a new merkle tree. `leaves.len()` should be power of two.
+    #[cfg(not(feature = "parallel"))]
     pub fn new<L: Borrow<P::Leaf>>(
         leaf_hash_param: &LeafParam<P>,
         two_to_one_hash_param: &TwoToOneParam<P>,
         leaves: impl IntoIterator<Item = L>,
     ) -> Result<Self, crate::Error> {
-        let mut leaves_digests = Vec::new();
+        // Serial version of new
+        // Compute and store hash values for each leaf
+        let leaves_digest: Result<Vec<_>, _> = leaves
+            .into_iter()
+            .map(|leaf| P::LeafHash::evaluate(leaf_hash_param, leaf))
+            .collect();
+
+        Self::new_with_leaf_digest(leaf_hash_param, two_to_one_hash_param, leaves_digest?)
+    }
+
+    /// Returns a new merkle tree. `leaves.len()` should be power of two.
+    #[cfg(feature = "parallel")]
```

Comment:
Maybe you can avoid having a separate block by using `cfg_into_iter`: https://github.com/arkworks-rs/std/blob/master/src/lib.rs#L136
See example of usage here: https://github.com/arkworks-rs/algebra/blob/ab13aa09ae3c11cde0224028dee7b878bbcf9246/ec/src/models/mnt6/mod.rs#L56